examples/op-errors/src/main.rs

```diff
@@ -49,10 +49,13 @@ where
         .iter()
         .map(|port| {
             ReceiveBatch::new(port.clone())
-                .map(|p| p.parse::<Ethernet>())
-                .filter(|p| p.ether_type() == EtherTypes::Ipv6)
+                .filter(|p| {
+                    let ethernet = p.peek::<Ethernet>().unwrap();
+                    ethernet.ether_type() == EtherTypes::Ipv6
+                })
                 .map(|p| {
-                    let v6 = p.parse::<Ipv6>()?;
+                    let ethernet = p.parse::<Ethernet>()?;
```

Comment:
efficiency wise the old example is actually better cuz ethernet is not parsed twice. 😄

Reply:
sure, i just wanted to showcase the example :).

Reply:
i'll add a comment that this is for showcasing "how peek works"

Reply:
ah, will show this on ipv4or6.